sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample12_ConsumeEventsWithEventProcessor.cs

```diff
@@ -40,102 +40,105 @@ public class Sample12_ConsumeEventsWithEventProcessor : IEventHubsSample
             {
-                // An event processor is associated with a specific Event Hub and a consumer group.  It receives events from
-                // multiple partitions in the Event Hub, passing them to the user for processing.
-                //
-                // A partition manager may create checkpoints and list/claim partition ownership.  The user can implement their
-                // own partition manager by creating a subclass from the PartitionManager abstract class.  Here we are creating
-                // a new instance of an InMemoryPartitionManager, provided by the Azure.Messaging.EventHubs.Processor namespace.
-                // This isn't relevant to understanding this sample, but is required by the event processor constructor.
-
-                PartitionManager partitionManager = new InMemoryPartitionManager();
-
-                // It's also possible to specify custom options upon event processor creation.  We want to receive events from
-                // the latest available position so older events don't interfere with our sample.  We also don't want to wait
-                // more than 1 second for every set of events.
-
-                EventProcessorOptions eventProcessorOptions = new EventProcessorOptions
-                {
-                    InitialEventPosition = EventPosition.Latest,
-                    MaximumReceiveWaitTime = TimeSpan.FromSeconds(1)
-                };
-
-                // Let's finally create our event processor.  We're using the default consumer group that was created with the Event Hub.
+                DefaultInitialEventPosition = EventPosition.Latest,
+                MaximumReceiveWaitTime = TimeSpan.FromSeconds(1)
+            };
 
-                var eventProcessor = new EventProcessor(EventHubConsumer.DefaultConsumerGroupName, client, partitionManager, eventProcessorOptions);
+            // Let's finally create our event processor.  We're using the default consumer group that was created with the Event Hub.
 
+            await using (var eventProcessor = new EventProcessorClient(EventHubConsumerClient.DefaultConsumerGroupName, partitionManager, connectionString, eventHubName, eventProcessorOptions))
+            {
                 int totalEventsCount = 0;
                 int partitionsBeingProcessedCount = 0;
 
                 // TODO: explain callbacks setup once the public API is finished for the next preview.
 
                 eventProcessor.InitializeProcessingForPartitionAsync = (PartitionContext partitionContext) =>
                 {
-                    // This is the last piece of code guaranteed to run before event processing, so all initialization
-                    // must be done by the moment this method returns.
+                        // This is the last piece of code guaranteed to run before event processing, so all initialization
+                        // must be done by the moment this method returns.
```

Comment:
For some reason, the code in every handler has indentation issues.

Reply:
Yes it does.   That was weird.